### PR TITLE
fix(sdk): make modals use the correct portal

### LINKED ISSUE #50521: [SDK] modals don't use the correct portal on storybook and cypress component tests, this leads to wrong styles being applied
https://metaboat.slack.com/archives/C063Q3F1HPF/p1732639135868419

This is making CI break and therefore blocking https://github.com/metabase/metabase/pull/50466 

Cypress:
start from interactive, drill drown on anything with "See these records" -> save, select a destination

Image

Storybook:

Image

### PR DESCRIPTION
**Pull request by npretto**
Closes https://github.com/metabase/metabase/issues/50521

This fixes two issues:
- when theme was not passed the mantine modals were not using the portal container correctly (fixes #50521), fixed in `SdkThemeProvider.tsx`
- "legacy modals" were being added to the body, not to our modal container, fixed in WindowModal

### CODE REVIEW

File: e2e/test-component/scenarios/embedding-sdk/styles-tests.cy.spec.tsx [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
+import { describeEE, modal, updateSetting } from "e2e/support/helpers";
 import {
   DEFAULT_SDK_PROVIDER_CONFIG,
   mockAuthProviderAndJwtSignIn,
   signInAsAdminAndEnableEmbeddingSdk,
 } from "e2e/support/helpers/component-testing-sdk";
 import { getSdkRoot } from "e2e/support/helpers/e2e-embedding-sdk-helpers";
 
-describeEE("scenarios > embedding-sdk > static-dashboard", () => {
+describeEE("scenarios > embedding-sdk > styles", () => {
```
Comment (npretto) [inline e2e/test-component/scenarios/embedding-sdk/styles-tests.cy.spec.tsx:17]: this was probably a copy paste mistake, this file is not related to static dashboards

**Review by WiNloSt — APPROVED** [review]
the tests seem to work as expected.

**Comment by npretto** [comment]
@metabase-bot backport release-x.51.x
